src/test/java/be/seeseemelk/mockbukkit/UnsafeValuesTest.java

```diff
@@ -157,4 +159,93 @@ void fromLegacy_MaterialData_Null_ReturnsNull()
 		assertThrows(NullPointerException.class, () -> mockUnsafeValues.fromLegacy((MaterialData) null));
 	}
 
+	@Test
+	void testMaterialThatIsOnlyItemTranslationKey()
```

Comment:
You have multiple tests in one test. Either make this a parameterized test (with for example a methodsource) or separate the tests. Same applies to the other testcases

Reply:
I tried grouping them together to be interpretable in a meaningful way. 
```java
assertEquals("block.minecraft.stone", mockUnsafeValues.getBlockTranslationKey(Material.STONE));
assertEquals("block.minecraft.stone", mockUnsafeValues.getItemTranslationKey(Material.STONE));
```
This clearly shows that both translation keys result in the same value.
```java
assertEquals("block.minecraft.acacia_sign", mockUnsafeValues.getBlockTranslationKey(Material.ACACIA_SIGN));
assertEquals("block.minecraft.acacia_sign", mockUnsafeValues.getBlockTranslationKey(Material.ACACIA_WALL_SIGN));
```
This clearly shows that a material and its "wall" counterpart material have the same translation key.

I worry that this readability is lost when extracting them into single tests.

Reply:
Look into Parameterized tests for Junit

Reply:
How can I assert something programmatically without relying on string names?
```java
	@ParameterizedTest
	@EnumSource(value = Material.class, names = {"STONE", "DIRT", "WHEAT", "NETHER_WART"})
	void testMaterialThatIsItemAndBlockTranslationKey(Material material)
	{
		assertEquals("block.minecraft." + material.toString().toLowerCase(), mockUnsafeValues.getBlockTranslationKey(material));
	}
```
I don't really like this because I am relying on the string material name. This becomes especially complicated for the signs, where the material enum string is not correct.
Maybe I will just write a million separate tests.

Reply:
Try methodsource instead. Something like this should work:
```java


@ParameterizedTest
@MethodSource("stringIntAndListProvider")
void testWithMultiArgMethodSource(String str, int num, List<String> list) {
    assertEquals(5, str.length());
    assertTrue(num >=1 && num <=2);
    assertEquals(2, list.size());
}

static Stream<Arguments> stringIntAndListProvider() {
    return Stream.of(
        arguments("apple", 1, Arrays.asList("a", "b")),
        arguments("lemon", 2, Arrays.asList("x", "y"))
    );
}
```

Reply:
Do you have a one-assert-per-test rule or were your comments just suggesting I should split the test cases up even further?
Because if you don't then I can split the tests up a little bit.

Reply:
There's no rule for this. I do think it's good practice to split the tests as much as possible. Why? Because in your case, if you have multiple checks that are independent of each other in the same test (that's why I'm saying it's multiple tests in the same testcase), if one of them fails the rest of the test will not run, therefore not providing information on whether those will fail or not.

Also please use a parameterized test, it's a cleaner way to do tests that are very similar to each other. (Sorry for being a bit over the place here, btw, I retract my statement about splitting the tests, paramterized tests are that much better).